builtin/credential/aws/backend.go

```diff
@@ -18,7 +17,13 @@ import (
 	cache "github.com/patrickmn/go-cache"
 )
 
-var defaultAllowedSTSRequestHeaders = []string{"Authorization", "Content-Length", "Content-Type", "User-Agent", "X-Amz-Date", textproto.CanonicalMIMEHeaderKey(iamServerIdHeader)}
+const amzHeaderPrefix = "X-Amz-"
+var defaultAllowedSTSRequestHeaders = []string{
+	"X-Amz-Date",
```

Comment:
Should we also include `X-Amz-Credential`, based on the common params for STS from [this page](https://docs.aws.amazon.com/STS/latest/APIReference/CommonParameters.html)?